WcaOnRails/app/views/competitions/_competition_form.html.erb

```diff
@@ -261,87 +261,52 @@
         <%= f.input :base_entry_fee_lowest_denomination, as: :money_amount, currency: @competition.currency_code, disabled: disable_money_input, currency_selector: "#competition_currency_code" %>
 
         <script>
-          var baseFeeUSD = -1;
-
-          // Note: unfortunately we cannot trigger a change on the currency
-          // code update, because we don't have access to the currency
-          // "subunit_to_unit" from the js world.
           function UpdateDues(){
-            <% countryBand = CountryBand.where(iso2: @competition.country_iso2).map(&:number)[0] %>
-            <% if countryBand.present? && countryBand > 0 %>
-              var bandFeeUSD = <%= CountryBand::BANDS[countryBand][:value] %>;
-            <% else %>
-              var bandFeeUSD = 0;
-            <% end %>
-            var pricePerCompetitor = Math.max(bandFeeUSD, baseFeeUSD) * 100;
-            var compLimit = $('select[name="competition[competitor_limit_enabled]"]').val() === "true"
-              ? $('input[name="competition[competitor_limit]"]').val() : null;
-            window.wca.cancelPendingAjaxAndAjax('currency_convert', {
-              url: '<%= currency_convert_path %>',
+
+            var obj = $("#competition-form").serializeJSON();
+
+            var compLimitEnabled = obj['competition[competitor_limit_enabled]']
+            var compLimit = compLimitEnabled === "true" ? obj['competition[competitor_limit]'] : null
+            var countryId = obj['competition[countryId]']
+            var currencyCode = obj['competition[currency_code]']
+            var entryFee = obj['competition[base_entry_fee_lowest_denomination]']
+
+            window.wca.cancelPendingAjaxAndAjax('calculate_dues', {
+              url: '<%= calculate_dues_path %>',
               data: {
-                value: compLimit === null ? pricePerCompetitor : pricePerCompetitor * compLimit,
-                from_currency: "USD",
-                to_currency: "<%= @competition.currency_code %>",
+                competitor_limit_enabled: compLimitEnabled,
+                competitor_limit: compLimit,
+                country_id: countryId,
+                currency_code: currencyCode,
+                entry_fee_cents: entryFee,
```

Comment:
I think you can inline all of these variables like so:
```js
data: {
  competitor_limit_enabled: obj['competition[competitor_limit_enabled]']
  etc...
}
```

Reply:
`compLimitEnabled`, `compLimit`, and `currencyCode` are used later in the HTML injection. Is it better to keep those variables or reference them again through `obj[...]`?

Reply:
That's very debatable and mostly depends on taste. I'm fine with the current solution as it hopefully won't stick around for long.